projects/control-service/projects/base/src/main/java/com/vmware/taurus/security/SecurityConfiguration.java

```diff
@@ -179,4 +204,67 @@ public JwtDecoder jwtDecoder() {
         }
         return Collections.emptySet();
     }
+
+    /*
+        KERBEROS config settings, a lot of these are optional.
+     */
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) {
+        if (enableKRBAuth) {
+            auth.authenticationProvider(kerberosServiceAuthenticationProvider());
+        }
+    }
+
+    @Bean
+    @ConditionalOnProperty(value = "datajobs.security.kerberos.enabled")
+    public SpnegoAuthenticationProcessingFilter spnegoAuthenticationProcessingFilter(
+          AuthenticationManager authenticationManager) {
+        SpnegoAuthenticationProcessingFilter filter = new SpnegoAuthenticationProcessingFilter();
+        filter.setAuthenticationManager(authenticationManager);
+        return filter;
+    }
+
+    @Bean
+    @ConditionalOnProperty(value = "datajobs.security.kerberos.enabled")
+    public KerberosServiceAuthenticationProvider kerberosServiceAuthenticationProvider() {
+        KerberosServiceAuthenticationProvider provider = new KerberosServiceAuthenticationProvider();
+        provider.setTicketValidator(sunJaasKerberosTicketValidator());
+        provider.setUserDetailsService(dataJobsUserDetailsService());
+        return provider;
+    }
+
+    @Bean
+    @ConditionalOnProperty(value = "datajobs.security.kerberos.enabled")
+    public SunJaasKerberosTicketValidator sunJaasKerberosTicketValidator() {
+        SunJaasKerberosTicketValidator ticketValidator = new SunJaasKerberosTicketValidator();
+        ticketValidator.setServicePrincipal(kerberosPrincipal);
+        ticketValidator.setKeyTabLocation(new FileSystemResource(keytabFileLocation));
+        ticketValidator.setDebug(true);
+        return ticketValidator;
+    }
+
+    @Bean
+    @ConditionalOnProperty(value = "datajobs.security.kerberos.enabled")
+    public SecurityConfiguration.DataJobsUserDetailsService dataJobsUserDetailsService() {
+        return new SecurityConfiguration.DataJobsUserDetailsService();
+    }
+
+    @Override
+    @Bean
+    @ConditionalOnProperty(value = "datajobs.security.kerberos.enabled")
```

Comment:
Extract datajobs.security.kerberos.enabled as contant. 

Also there's class called FeatureFlags, I think those kinda (feature on/off) flags should be defined there

Reply:
ConditionalOnProperty is effectively doing that. It will return false if the property is not present in the environment by default.
 I think using a Conditional annotation is the best way to configure those Beans (it's managed by spring) I'm worried that if KRB auth is disabled those effectively blank configurations could interfere with other authenticators plus it's neater than having a bunch of if statements in every bean.
 I don't think there is an option like ConditionalOnBoolean and the other option of using a boolean in those Beans doesn't look ideal to me.

Reply:
I mean you can put it as a constant there so that the string is re-used

Reply:
For some reason the code won't compile when I put use a constant from the FeatureFlags class in an annotation in the SecurityConfiguration class so I put the constant in the same class.